api/src/test/java/org/openmrs/api/CohortServiceTest.java

```diff
@@ -644,5 +654,195 @@ public void getMemberships_shouldGetMembershipsAsOfADate() throws Exception {
 		assertFalse(membersAsOfDate.isEmpty());
 		assertTrue(membersAsOfDate.stream().anyMatch(m -> m.getStartDate().equals(dateToTest)));
 	}
-
+	
+	/**
+	 * @verifies {@link Cohort#getMemberships(Date)}
+	 */
+	@Test
+	@Verifies(value = "should not get matching memberships of a cohort as of a date", method = "getMemberships(Date)")
+	public void getMemberships_shouldNotGetMatchingMembershipsAsOfADate() throws Exception {
+		executeDataSet(COHORT_XML);
+		
+		Cohort cohort = Context.getCohortService().getCohort(1);
+		
+		CohortMembership newMember = new CohortMembership((new Patient(4)));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-11-01 00:00:00");
+		newMember.setStartDate(startDate);
+		service.addMembershipToCohort(cohort, newMember);
+		
+		Date dateToTest = dateFormat.parse("2016-11-01 00:00:00");
+		List<CohortMembership> membersAsOfDate = cohort.getMemberships(dateToTest);
+		assertTrue(membersAsOfDate.isEmpty());
+		assertFalse(membersAsOfDate.stream().anyMatch(m -> m.getStartDate().equals(dateToTest)));
+	}
+	
+	/**
+	 * @verifies {@link CohortMembership#isMembershipActive(Date)}
+	 */
+	@Test
+	@Verifies(value = "membership should be active as of a date", method = "isMembershipActive(Date)")
+	public void isMembershipActive_shouldBeActiveAsOfDate() throws Exception {
+		CohortMembership newMember = new CohortMembership(new Patient(4));
+		SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+		Date startDate = dateFormat.parse("2017-01-01 00:00:00");
+		newMember.setStartDate(startDate);
+		
+		Date dateToTest = dateFormat.parse("2017-01-01 12:00:00");
+		assertTrue(newMember.isMembershipActive(dateToTest));
+		assertTrue(newMember.isMembershipActive(new Date()));
+		assertTrue(newMember.isMembershipActive(startDate));
+	}
+	
+	/**
+	 * @verifies {@link CohortMembership#isMembershipActive(Date)}
+	 */
+	@Test
+	@Verifies(value = "membership should be inactive as of a date", method = "isMembershipActive(Date)")
+	public void isMembershipActive_shouldBeInactiveAsOfDate() throws Exception {
```

Comment:
Method name is not intuitive, should say something like shouldReturnFalseIfStartDateIsAfterAsOfDate()